File: AddressBook/AddressBook/Dtos/PersonPhone.cs
using AddressBook.Entities;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace AddressBook.Dtos
{
    public class PersonPhoneRes
    {
        public int PersonPhoneId { get; set; }
        public string? Phone { get; set; }

        public int? CountryId { get; set; }
        public Country? Country { get; set; }

        public int? PersonId { get; set; }
        public PersonRes? Person { get; set; }

        public int? PhoneLabelId { get; set; }
        public PhoneLabelRes? PhoneLabel { get; set; }
    }

    public class PersonPhoneReqEdit
    {
        [Required]
        public string? Phone { get; set; }
        // Foreign keys
        public int? CountryId { get; set; }
        public int? PersonId { get; set; }
        public int? PhoneLabelId { get; set; }
    }

    public class PersonPhoneReqSearch
    {
        [Required]
        public int? PersonId { get; set; }
    }
}

File: AddressBook/react-client/src/dtos/PersonPhone.ts
import Common from "../utility/Common";
import { CountryRes } from "./Country";
import { PagedReq } from "./PagedReq";
import { PersonRes } from "./Person";
import { PhoneLabelRes } from "./PhoneLabel";

export interface PersonPhoneRes {
  personPhoneId?: number;
  phone?: string;
  countryId?: number;
  country?: CountryRes;
  personId?: number;
  person?: PersonRes;
  phoneLabelId?: number;
  phoneLabel?: PhoneLabelRes;
}

export class PersonPhoneReqEdit {
  phone?: string;
  countryId?: number = 0;
  personId?: number = 0;
  phoneLabelId?: number = 0;
}

export class PersonPhoneReqSearch extends PagedReq {
  personId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy,
      searchText = "",
    }: PagedReq,
    {personId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.personId = personId;
  }
}